File: C4A_android/Android/app/src/main/java/com/example/code4all/error/ErrorNetwork.java
package com.example.code4all.error;

import android.content.Context;
import com.android.volley.*;
import com.example.code4all.R;
import com.example.code4all.serverhandler.ServerHandler;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.UnsupportedEncodingException;
import java.nio.charset.StandardCharsets;

public class ErrorNetwork extends VolleyError {

//    Keys for json search
    private final String ERROR_MESSAGE_KEY = "error";
    private final String ERROR_CODE_KEY = "errorCode";

    private JSONObject errorContent;


    public ErrorNetwork(VolleyError volleyError, Context context) throws JSONException {
        errorContent = new JSONObject();
        errorContent.put(ERROR_MESSAGE_KEY, setMessage(volleyError, context));

//            int statusCode = volleyError.networkResponse.statusCode;
//            if (volleyError.networkResponse.data != null) {
//                JSONObject dataJsonMessage = new JSONObject(new String(volleyError.networkResponse.data, "UTF-8"));
//                errorContent.put(ERROR_MESSAGE_KEY, ServerHandler.getStringFromJsonObject(dataJsonMessage, ERROR_MESSAGE_KEY));
//            } else {
//                errorContent.put(ERROR_MESSAGE_KEY, "Check your internet connection");
//            }

    }

    private String setMessage(VolleyError error, Context context) {
        /*
        if (error.networkResponse != null) {
            if (error.networkResponse.data != null) {
                JSONObject dataJsonMessage = new JSONObject(new String(error.networkResponse.data, StandardCharsets.UTF_8));
                return ServerHandler.getStringFromJsonObject(dataJsonMessage, ERROR_MESSAGE_KEY);
            }
        }*/
        if (error instanceof NetworkError) {
             return context.getString(R.string.network_error_network_error);
        } else if (error instanceof ServerError) {
            return context.getString(R.string.network_error_server_error);
        } else if (error instanceof AuthFailureError) {
            return context.getString(R.string.network_error_auth_failure);
        } else if (error instanceof ParseError) {
            return context.getString(R.string.network_error_parser_error);
        } else if (error instanceof TimeoutError) {
            return context.getString(R.string.network_error_time_out);
        }
        return "";
    }
    public String diplayErrorMessage(){
        try {
            return ServerHandler.getStringFromJsonObject(errorContent, ERROR_MESSAGE_KEY);
        } catch (JSONException e) {
            e.printStackTrace();
        }

        return "";
    }

    public String getErrorCode() throws JSONException {
        return ServerHandler.getStringFromJsonObject(errorContent, ERROR_CODE_KEY);
    }
}

File: C4A_android/Android/app/src/main/java/com/example/code4all/settings/ErrorNetwork.kt
package com.example.code4all.settings

import android.content.Context
import com.android.volley.*
import com.example.code4all.R
import com.example.code4all.serverhandler.ServerHandler
import org.json.JSONException
import org.json.JSONObject
import java.io.UnsupportedEncodingException


const val ERROR_MESSAGE_KEY = "error"
const val ERROR_CODE_KEY = "errorCode"

class ErrorNetwork @Throws(UnsupportedEncodingException::class, JSONException::class)
constructor(volleyError: VolleyError, context: Context?) {

    private val errorContent: JSONObject = JSONObject()

    val errorCode: String
        @Throws(JSONException::class)
        get() = ServerHandler.getStringFromJsonObject(errorContent, ERROR_CODE_KEY)

    init {
        errorContent.put(ERROR_MESSAGE_KEY, setMessage(volleyError, context))
    }

    @Throws(UnsupportedEncodingException::class, JSONException::class)
    private fun setMessage(error: VolleyError, context: Context?): String {
        if (error.networkResponse != null) {
            if (error.networkResponse.data != null) {
                val dataJsonMessage = JSONObject(String(error.networkResponse.data))
                return ServerHandler.getStringFromJsonObject(dataJsonMessage, ERROR_MESSAGE_KEY)
            }
        }

        context?.let {
            return when (error) {
                is NetworkError -> context.getString(R.string.network_error_network_error)
                is ServerError -> context.getString(R.string.network_error_server_error)
                is AuthFailureError -> context.getString(R.string.network_error_auth_failure)
                is ParseError -> context.getString(R.string.network_error_parser_error)
                is TimeoutError -> context.getString(R.string.network_error_time_out)
                else -> ""
            }
        }
        return ""
    }

    fun getErrorMessage(): String {
        try {
            return ServerHandler.getStringFromJsonObject(errorContent, ERROR_MESSAGE_KEY)
        } catch (e: JSONException) {
            e.printStackTrace()
        }

        return ""
    }
}
